client/components/data/query-jetpack-scan/index.tsx

```diff
@@ -23,7 +23,7 @@ const QueryJetpackScan = ( { siteId }: Props ) => {
 			return;
 		}
 		siteId && dispatch( requestJetpackScanStatus( siteId ) );
-	}, [ siteId ] );
+	}, [ siteId, dispatch, requestingJetpackScan ] );
```

Comment:
We have to rethink this a bit. Adding `requestingJetpackScan` as a dependency makes the effect to be executed every time its value changes. At first, it will be `'pending'`, then it will change to `'success'` or `'failure'` and it will trigger the execution of the effect once more. This will keep happening regardless of the result of the request.

If we remove the dependency, it will get executed only once unless we add a retry mechanism (I believe we have examples of that). @enejb, what do you think? 

Here we can see that the effect is triggered several times.
![image](https://user-images.githubusercontent.com/3418513/79395870-fdc6d900-7f50-11ea-9ace-20f97482a9d2.png)

Reply:
Oh, good to know! This part of my PR was only a result of me following advice from the linter, and I wasn't sure if the repeating behavior was intentional or not.

We can definitely take that out if it has undesirable effects.

Reply:
I think we should be need to have fine grain control over when we request the api. 

For example if things are fine (No threats are found) 
We should just call the api once. The query component might be triggerd by the sidebar or the main Scan component. 

But if we are in the process of scanning we want to be pulling the api every few 15-30 seconds to get the latest update